### PR TITLE
Added object-deep-map-values

### PR DESCRIPTION
**Pull request by maxsynnott**
Added function to deeply map an objects values

example:
```js
import deepMapValues from "just-deep-map-values";
const squareFn = (number) => number * number;
deepMapValues({ a: 1, b: { c: 2, d: { e: 3 } } }, squareFn); // => { a: 1, b: { c: 4, d: { e: 9 } } }
```

### CODE REVIEW

File: packages/object-deep-map-values/index.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module.exports = deepMapValues;
+
+function deepMapValues(obj, fn) {
+  return Object.keys(obj).reduce(function(acc, key) {
```
Comment (angus-c) [inline packages/object-deep-map-values/index.js:4]: Can you replace `reduce` with a for loop? Current guidelines are to treat all utilities as potentially hot code and so micro-optimize accordingly. (Functions that include deep object traversal could be particularly susceptible to the effect of multiple function calls, specially in non-optimized browsers)

File: test/object-deep-map-values/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const test = require('../util/test')(__filename);
+const deepMapValues = require('../../packages/object-deep-map-values');
+
+const squareFn = (value) => value * value;
+const concatFn = (value, key) => key + value;
+
+test('correctly maps using value argument', (t) => {
```
Comment (angus-c) [inline test/object-deep-map-values/index.js:7]: Please add tests for a) a flat object with no nesting b) failing cases (e.g missing arguments or arguments with wrong types).

**Review by angus-c — COMMENTED** [review]
Thanks! Couple of things.

File: packages/object-deep-map-values/index.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module.exports = deepMapValues;
+
+function deepMapValues(obj, fn) {
```
Comment (angus-c) [inline packages/object-deep-map-values/index.js:3]: I know I didn't do this for object-map-keys and other early just utils but most Just utils now check argument types and throw if they are incorrect or missing (for those not using TS)
(example https://github.com/angus-c/just/blob/master/packages/array-mean/index.js#L8-L16)

File: packages/object-deep-map-values/index.js (hunk 1 above)
Comment (maxsynnott) [reply packages/object-deep-map-values/index.js]: Done

File: test/object-deep-map-values/index.js (hunk 2 above)
Comment (maxsynnott) [reply test/object-deep-map-values/index.js:7]: a: This was already being tested
b: Added

File: packages/object-deep-map-values/index.js (hunk 3 above)
Comment (maxsynnott) [reply packages/object-deep-map-values/index.js:3]: Agreed this should be done.

Added

File: packages/object-deep-map-values/index.mjs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+var deepMapValues = function deepMapValues(obj, fn) {
```
Comment (angus-c) [inline packages/object-deep-map-values/index.mjs:1]: need to re-run build step to pickup removal of reduce.

**Review by angus-c — COMMENTED** [review]
Thanks for the changes, I'm still getting seeing lint errors around use of double quotes in index.js (unless github is messing with me)

**Comment by maxsynnott** [comment]
> Thanks for the changes, I'm still getting seeing lint errors around use of double quotes in index.js (unless github is messing with me)

My bad, fixed again. So used to prettier handling it for me 😬

**Comment by PuruVJ** [comment]
I need to confirm, is this function part of the `collection` submodule?

**Comment by maxsynnott** [comment]
> I need to confirm, is this function part of the `collection` submodule?

No, my understanding is it should be part of the `object` submodule.

**Comment by PuruVJ** [comment]
Ah ok. @angus-c heads up to add this to the default README

**Comment by maxsynnott** [comment]
> Ah ok. @angus-c heads up to add this to the default README

This will be resolved by #381

**Pull request merged**